### PR TITLE
Simplify some `Relation.Binary` imports

### PR DESCRIPTION
**Pull request by Saransh-cpp**
This PR does not get rid of all the `Relation.Binary` imports. I'll be creating PRs for the same.

### CODE REVIEW

File: src/Relation/Binary/Indexed/Homogeneous/Structures.agda [hunk 1]
```diff
@@ -20,7 +20,10 @@ module Relation.Binary.Indexed.Homogeneous.Structures
 open import Data.Product using (_,_)
 open import Function.Base using (_⟨_⟩_)
 open import Level using (Level; _⊔_; suc)
-open import Relation.Binary as B using (_⇒_)
+open import Relation.Binary.Core using (_⇒_)
+import Relation.Binary.Definitions as Definitions
+import Relation.Binary.Bundles as Bundles
+import Relation.Binary.Structures as Structures
```
Comment (Saransh-cpp) [inline src/Relation/Binary/Indexed/Homogeneous/Structures.agda:26]: I am not very sure about the naming schemes here

Comment (gallais) [reply src/Relation/Binary/Indexed/Homogeneous/Structures.agda:26]: Can't you import them all as `B`?

Comment (Saransh-cpp) [reply src/Relation/Binary/Indexed/Homogeneous/Structures.agda]: That worked, thanks @gallais!

**Review by JacquesCarette — APPROVED** [review]

File: src/Text/Regex/Properties.agda [hunk 2]
```diff
@@ -23,7 +23,8 @@ open import Relation.Nullary.Negation
   using (¬_; contradiction)
 
 import Relation.Unary  as U
-open import Relation.Binary using (Decidable)
+open import Relation.Binary.Definitions using (Decidable)
+import Relation.Binary.PropositionalEquality as P
```
Comment (Saransh-cpp) [inline src/Text/Regex/Properties.agda:27]:
```suggestion
```

**Comment by Taneb** [comment]
Does anyone have any objection to merging this and the similar PRs? I think they've sat open long enough

**Comment by JacquesCarette** [comment]
I'd like some truly independent person to do a final review (perhaps @gallais ?) before doing so.

**Review by gallais — APPROVED** [review]

**Pull request merged**